### ISSUE TITLE
Replace default datasource by HikariCP

### ISSUE THREAD

**Issue by ppodgorsek** [ppodgorsek/sap-commerce-utils#3]
HikariCP is much faster than the default datasource in SAP Commerce, it should therefore replace it.

**Comment by ppodgorsek** [comment]
Done

**Comment by tobijdc** [comment]
Nice
Thank you for open sourceing your findings and ideas.

A few questions:
Have you been using this in production in any way and did you see a difference (for example in initialization time)?
Are existing configuration properties within SAP Commerce taken into account or have some HikariCP ones have to be used instead?
Having this in the Readme (or in the Wiki) might be valuable for others.

Thanks again

**Comment by ppodgorsek** [comment]
Some database properties from Commerce are currently not supported in this version, such as the table prefix. I will create additional enhancement tickets to test/document those too, for each database type.

Regarding the performance, we have seen improvements in the Backoffice navigation locally and on lower environments. Locally, the initialisation time was also reduced by 25%. (we only run updates on other environments)

Performance tests of our whole application are planned in 10 days, which should give us a much better idea of the scale of the improvement across all areas. Once done, I will provide more information here.

**Comment by ppodgorsek** [comment]
A quick update on performance tests: on the contrary to what we were expecting, we didn't see major performance improvements during our tests regarding response times for example.

We did however notice a bit improvement in the JVM garbage collection cycles during stress tests, the various memory spaces remaining much more stable. We are using the G1 garbage collector.

Focusing on garbage collection CPU times, we went from 0.45% of the CPU time spent in GC to 0.15% on average. Stop-the-world operations have become incredibly rare.
The cycles of the eden, survivor and old gen spaces all behave in a much more regular way.
